api/handler.go

```diff
@@ -151,8 +151,9 @@ func streamResults(results chan *events.Event, w http.ResponseWriter, r *http.Re
 	for {
 		select {
 		case <-cn.CloseNotify():
-			return nil
-		case ev := <-results:
+			eventChannel.Stop()
+			return m.CloseDeadChannels()
```

Comment:
We need to keep on waiting for data here otherwise we'll deadlock (if no one is reading while the writer wants to write)